c10/core/Scalar.h

```diff
@@ -35,6 +35,12 @@ class C10_API Scalar {
 
 #undef DEFINE_IMPLICIT_CTOR
 
+template<typename T>
+Scalar(typename std::enable_if<std::is_same<T, bool>::type>::type, bool vv)
```

Comment:
does this work?  Have you tried calling it this way in C++?

Look at the examples from https://en.cppreference.com/w/cpp/types/enable_if -- they put the `enable_if` in the template.

Reply:
Fixed and tested [via](https://github.com/izdeby/playground/blob/dedf311e866d4b124081e881c522e32dab86123d/C%2B%2B/enableIfTest.cpp).